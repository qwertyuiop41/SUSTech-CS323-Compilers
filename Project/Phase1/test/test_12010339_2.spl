int test(int a, int b, int c) 
{

  if(a<b/c){
    return b*c;
  }
  
  if(a>b+c){
    return b-c;
  }
  else{
    return a;
  }
  return a*b*c-c/a;
  
}