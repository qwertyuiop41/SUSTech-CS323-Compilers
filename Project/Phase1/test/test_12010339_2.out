Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: test
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: a
          COMMA
          VarList (1)
            ParamDec (1)
              Specifier (1)
                TYPE: int
              VarDec (1)
                ID: b
            COMMA
            VarList (1)
              ParamDec (1)
                Specifier (1)
                  TYPE: int
                VarDec (1)
                  ID: c
        RP
      CompSt (2)
        LC
        StmtList (4)
          Stmt (4)
            IF
            LP
            Exp (4)
              Exp (4)
                ID: a
              LT
              Exp (4)
                Exp (4)
                  ID: b
                DIV
                Exp (4)
                  ID: c
            RP
            Stmt (4)
              CompSt (4)
                LC
                StmtList (5)
                  Stmt (5)
                    RETURN
                    Exp (5)
                      Exp (5)
                        ID: b
                      MUL
                      Exp (5)
                        ID: c
                    SEMI
                RC
          StmtList (8)
            Stmt (8)
              IF
              LP
              Exp (8)
                Exp (8)
                  ID: a
                GT
                Exp (8)
                  Exp (8)
                    ID: b
                  PLUS
                  Exp (8)
                    ID: c
              RP
              Stmt (8)
                CompSt (8)
                  LC
                  StmtList (9)
                    Stmt (9)
                      RETURN
                      Exp (9)
                        Exp (9)
                          ID: b
                        MINUS
                        Exp (9)
                          ID: c
                      SEMI
                  RC
              ELSE
              Stmt (11)
                CompSt (11)
                  LC
                  StmtList (12)
                    Stmt (12)
                      RETURN
                      Exp (12)
                        ID: a
                      SEMI
                  RC
            StmtList (14)
              Stmt (14)
                WHILE
                LP
                Exp (14)
                  Exp (14)
                    ID: a
                  GT
                  Exp (14)
                    Exp (14)
                      ID: b
                    PLUS
                    Exp (14)
                      ID: c
                RP
                Stmt (14)
                  CompSt (14)
                    LC
                    StmtList (15)
                      Stmt (15)
                        RETURN
                        Exp (15)
                          Exp (15)
                            ID: b
                          MINUS
                          Exp (15)
                            ID: c
                        SEMI
                    RC
              StmtList (17)
                Stmt (17)
                  RETURN
                  Exp (17)
                    Exp (17)
                      Exp (17)
                        Exp (17)
                          ID: a
                        MUL
                        Exp (17)
                          ID: b
                      MUL
                      Exp (17)
                        ID: c
                    MINUS
                    Exp (17)
                      Exp (17)
                        ID: c
                      DIV
                      Exp (17)
                        ID: a
                  SEMI
        RC
